card/mod/06_bootstrap/set/all/bootstrap/form.rb

```diff
@@ -1,15 +1,14 @@
 format :html do
   def button_tag content_or_options = nil, options = {}, &block
-    if block_given?
-      content_or_options[:class] ||= ''
-      content_or_options[:class] += ' btn btn-default'
-    else
-      options[:class] ||= ''
-      options[:class] += ' btn btn-default'
-    end
+    bootstrapify_button( block_given? ? content_or_options : options )
     super(content_or_options, options, &block)
   end
   
+  def bootstrapify_button options
+    situation = options.delete(:situation) || 'default'
+    options[:class] = [ options[:class], 'btn', "btn-#{situation}" ].compact*' '
+  end
```

Comment:
Ok, that makes more sense.  More of a modifier on the container classes, so more general.  We don't overload 'state' as much as 'context', maybe that would be better.  Is it bootstrap related?  I mean, it's their class semantics, so maybe :bootstrap_state, but just :state might be clear enough.